ext/rkerberos/kadm5.c

```diff
@@ -934,6 +958,75 @@ static VALUE rkadm5_randkey_principal(VALUE self, VALUE v_user){
   return INT2NUM(n_keys);
 }
 
+/**
+ * Parses an array or a single string containing database arguments for kerberos functions.
+ * Returns NULL if v_db_args is nil, otherwise returns a NULL-Terminated array of NULL-Terminated strings
+ */
+char** parse_db_args(VALUE v_db_args){
+  long array_length;
+  char** db_args;
+  switch(TYPE(v_db_args)){
+    case T_STRING:
+      db_args = (char **) malloc(2 * sizeof(char *));
+      db_args[0] = StringValueCStr(v_db_args);
+      db_args[1] = NULL;
+      break;
+    case T_ARRAY:
+      // Multiple arguments
+      array_length = RARRAY_LEN(v_db_args);
+      db_args = (char **) malloc(array_length * sizeof(char *) + 1);
+      for(long i = 0; i < array_length; ++i){
+        VALUE elem = rb_ary_entry(v_db_args, i);
+        Check_Type(elem, T_STRING);
+        db_args[i] = StringValueCStr(elem);
+      }
+      db_args[array_length] = NULL;
+      break;
+    case T_NIL:
+      db_args = NULL;
+      break;
+    default:
+      rb_raise(rb_eTypeError, "Need Single String or Array of Strings for db_args");
+  }
+  return db_args;
+}
+
+/**
+ * Add parsed db-args to principal entry
+ */
+void add_db_args(kadm5_principal_ent_rec* entry, char** db_args){
```

Comment:
No, I don't think so generally, I just noticed the warnings on this file when I went to build it.